tmt/checks/coredump.py

```diff
@@ -0,0 +1,288 @@
+            output = guest.execute(Command(
+                "coredumpctl", "list", "--no-legend", "--no-pager", f"--since={start_time}"
+                )
+                ).stdout
+
+            if not output:
+                return False
+
+            # Process each crash entry
+            has_crashes = False
+            for line in output.splitlines():
+                fields = line.split()
+
+                pid = fields[4]
+                sig = fields[7]
+                corefile = fields[8]
+                exe = fields[9].replace("/", "_")
+
+                # Get detailed info for this crash
+                crash_info = guest.execute(
+                    Command("coredumpctl", "info", "--no-pager", pid)
+                    ).stdout
+
+                if not crash_info:
+                    logger.debug(f"No crash info available for PID {pid}")
+                    continue
+
+                # Skip if this crash matches any ignore pattern
+                if self.ignore_patterns:
+                    ignored = False
+                    for pattern in self.ignore_patterns:
+                        if pattern.search(crash_info):
+                            logger.debug(
+                                f"Ignoring crash due to pattern '{pattern.pattern}': {crash_info}"
+                                )
+                            ignored = True
+                            break
+                    if ignored:
+                        continue
+
+                # Save the crash info
+                info_file = f"{check_files_path}/dump.{exe}_{sig}_{pid}.txt"
+                guest.execute(
+                    ShellScript(f"sh -c coredumpctl info --no-pager {pid} > {info_file}")
+                    )
+                logger.debug(f"Saved crash info to {info_file}")
+
+                # Try to save the coredump if available
+                if corefile not in ("none", "missing"):
+                    dump_file = f"{check_files_path}/dump.{exe}_{sig}_{pid}.core"
+                    try:
+                        guest.execute(
+                            ShellScript(f"coredumpctl dump --no-pager -o {dump_file} {pid}")
+                            )
+                        logger.debug(f"Saved coredump to {dump_file}")
+                    except tmt.utils.RunError as exc:
+                        logger.debug(f"Failed to save coredump for PID {pid}: {exc}")
+                else:
+                    logger.debug(f"Skipping dump for PID {pid}, corefile status: {corefile}")
+
+                # This is a non-ignored crash
+                has_crashes = True
+
+            return has_crashes
+
+        except tmt.utils.RunError as exc:
+            logger.debug(f"Failed to check for crashes: {exc}")
+            return False
+
+    def _check_coredump(
+            self, invocation: "TestInvocation", event: CheckEvent, logger: tmt.log.Logger
+            ) -> ResultOutcome:
```

Comment:
In the name of connecting dots and letting users in, would it make sense to return not just an outcome, but also the list of logs, i.e. info and coredump files created by `_check_for_crashes()`? I would imagine `_check_for_crashes()` and `_check_coredump()` would basically collect the paths, and add them to `logs` field of the eventual check result, together with the outcome.